cdap-ui/server/config/parser.js

```diff
@@ -55,28 +55,35 @@ async function extractUITheme() {
   const DEFAULT_CONFIG = {};
   const cdapConfig = await extractConfig('cdap');
   let cdapSiteContents;
-  try {
-    cdapSiteContents = await getCdapSiteContents(cdapConfig);
-  } catch (e) {
-    log.info(e.toString());
-    return DEFAULT_CONFIG;
-  }
 
-  try {
-    cdapSiteContents = JSON.parse(cdapSiteContents);
-    const uiThemeProperty = cdapSiteContents.find(property => property.name === uiThemePropertyName);
-    if (!uiThemeProperty) {
-      log.info(`Unable to find ${uiThemePropertyName} property`);
+  if (uiThemePropertyName in cdapConfig) {
+    return getUIThemeConfig(cdapConfig[uiThemePropertyName]);
+  } else {
+    // This will only happen in UI development mode, when we don't pass in the
+    // cConf to the node server.js command. In this case, do an API call to
+    // /v3/config/cdap to get cdap config contents
```

Comment:
Isn't that this file is for development? https://github.com/caskdata/cdap/blob/release/5.1/cdap-ui/server/config/development/cdap.json

Reply:
Should we add the `ui.theme.file` in there so that we don't need to maintain a code path for fetching it from backend?

For theme configuration, fetching from backend doesn't make a lot of sense.

Reply:
Yes, if we pass in the 'ui.theme.property' to that file then UI can pick it up from that file. @ajainarayanan suggested that we do the API call if we can't 'ui.theme.file' from the cdap config passed it, but what you suggest makes sense. Ajai do you have any comments on this?

Reply:
For UI theme, it doesn't really make sense to consult backend config. Consider that when backend is not running vs backend is running, the UI could look very different.

Reply:
@chtyim I just discussed with @ajainarayanan, will make the change now